### DISCUSSION TITLE
关于自定义Platform的问题

### DISCUSSION THREAD

**Discussion by FZerolight** [xmake-io/xmake#3841] [Q&A] [answered]
Hi，我目前在使用xmake+Keil MDK(armcc)进行一些尝试，按照文档使用`cross`作为platform之后，会默认将静态链接库的命名为lib$(name).a，输出二进制命名为$(name)，而在Keil MDK+ARMCC的语境下静态链接库默认的命名为$(name).lib，而输出二进制为$(name).axf。在阅读了几个issue之后发现目前xmake并不提供platform api，建议的使用是在target中使用is_plat进行判断，请问在使用is_plat来做自定义platform的时候，有办法解决上述命名问题吗？Thanks in advance

**Comment by waruqi** [comment]
is_plat("xxxxx") 随意定，不是内置的都是自定义平台

**Reply by FZerolight** [reply]
我的意思是，在`is_plat("xxx")`的解决方案中，是否有能完成类似内置platform中`set_formats("static", "$(name).lib")`这样的操作的方法？

**Reply by waruqi** [reply]
这个，你自己 target 里面 set_extension set_filename set_basename 也可以改，根据自己的 is_plat 判断后

**Reply by FZerolight** [reply]
我理解这三个API更改的是build结果的文件名一类吧？有办法修改xmake本身在构建过程中搜索静态链接库命名逻辑的方法吗？例如我现在有一个已经编译好的库xxxx.lib，如果我直接在target里写`add_links("xxxx")`xmake默认会去找`libxxxx.a`而不是`xxxx.lib`。因为考虑到需要同时支持已有的Keil工程，直接把lib名字改了不是很合适，现在我的workaround是在`before_link`里面去拷贝改名，看起来不够优雅。

**Reply by waruqi** [reply]
这个你只能  after_link 或者 after_build 后，自己改，或者 os.cp 下。。

**Reply by FZerolight** [reply]
好吧，那在未来的版本中有计划考虑开放Custom Platform API吗？

**Reply by waruqi** [reply, marked as answer]
暂时不考虑，platform api 暂时设计上 还不稳定，随时会改。。暂时不对外开放

**Reply by FZerolight** [reply]
理解了，非常感谢
